pkg/virtual/syncer/transformations/defaultsummarizing.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformations
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var _ SummarizingRules = (*DefaultSummarizingRules)(nil)
+var _ SummarizingRulesProvider = (*DefaultSummarizingRules)(nil)
+
+// DefaultSummarizingRules provides a default minimal implementation of [SummarizingRules].
+// It only adds a status field, which for now is always promoted (see comments below in the code)
+type DefaultSummarizingRules struct{}
+
+type field struct {
+	Path              []string
+	PromoteToUpstream bool
+}
+
+var _ FieldToSummarize = field{}
+
+func (f field) Key() string {
+	return strings.Join(f.Path, ".")
+}
+
+func (f field) Set(resource *unstructured.Unstructured, value interface{}) error {
+	return unstructured.SetNestedField(resource.UnstructuredContent(), value, f.Path...)
+}
+
+func (f field) Get(resource *unstructured.Unstructured) (interface{}, bool, error) {
+	return unstructured.NestedFieldCopy(resource.UnstructuredContent(), f.Path...)
+}
+
+func (f field) Delete(resource *unstructured.Unstructured) {
+	unstructured.RemoveNestedField(resource.UnstructuredContent(), f.Path...)
+}
+
+func (f field) CanPromoteToUpstream() bool {
+	return f.PromoteToUpstream
+}
+
+func (f field) IsStatus() bool {
```

Comment:
odd special case

Reply:
This one is still needed. But to make a bit less odd I added more documentation.